celery/canvas.py

```diff
@@ -1490,6 +1542,21 @@ def freeze(self, _id=None, group_id=None, chord=None,
         self.id = self.tasks.id
         return body_result
 
+    def stamp(self, visitor=None, **headers):
+        if visitor:
+            headers.update(visitor.on_group_start(self, **headers))
```

Comment:
This is the wrong thing to do here.
We discussed having an `on_chord_header_start`/`on_chord_header_end` which by default calls `on_group_start` but can be overridden if needed.
This implementation also overlooks the chord's body completely.

Reply:
added + on_chord_body